dev/devicelab/lib/tasks/perf_tests.dart

```diff
@@ -1474,10 +1474,10 @@ class DevToolsMemoryTest {
       final Map<String, dynamic> data = json.decode(
         file('$project/$_kJsonFileName').readAsStringSync(),
       ) as Map<String, dynamic>;
-      final List<Map<String, dynamic>> samples = (data['samples'] as Map<String, dynamic>)['data'] as List<Map<String, dynamic>>;
+      final List<dynamic> samples = (data['samples'] as Map<String, dynamic>)['data'] as List<dynamic>;
```

Comment:
Not sure I follow why this was needed to change. Below it looks like it is just recast as a Map<String, dynamic> with no checks in between, so what was the issue with the previous version?

Reply:
The previous version gave an error because the actual type of `data['samples']['data']` was `List<dynamic>` and not `List<Map<String, dynamic>>`, so just casting it wouldn't work because they are not the same or derived types. the `cast` call works because it casts each entry in the list to a Map<String, dynamic> instead of casting the list itself (which is still a `List<dynamic>`).

Reply:
Tricky. Thanks for the explanation and quick fix.

Reply:
There's no way you could have caught this in the initial review, because you couldn't have known the type of `data['samples']['data']` was `List<dynamic>` and not `List<Map<String, dynamic>>`, just like I didn't know (and the compiler didn't know) until runtime. It didn't run in presubmit because it was a devicelab test.